cardano-testnet/src/Testnet/Util/Assert.hs

```diff
@@ -71,30 +71,44 @@ newtype Kind = Kind
   { kind :: Text
   } deriving (Eq, Show)
 
-data TraceNodeIsLeader = TraceNodeIsLeader
-  { kind :: Text
-  , slot :: Int
-  } deriving (Eq, Show)
-
-instance FromJSON TraceNodeIsLeader where
-  parseJSON = Aeson.withObject "TraceNodeIsLeader" $ \v -> do
-    k <- v .: "val" >>= (.: "kind")
-    if k == "TraceNodeIsLeader"
-    then TraceNodeIsLeader k <$> (v .: "val" >>= (.: "slot"))
-    else fail $ "Expected kind was TraceNodeIsLeader, found " <> show k <> "instead"
+data TraceNode
+  = TraceNode
+    { isLeader :: !Bool
+    , kind     :: !Text
+    , slot     :: !Int
+    }
+  deriving (Eq, Show)
+
+instance FromJSON TraceNode where
+  parseJSON = Aeson.withObject "TraceNode" $ \v -> do
+    kind' <- v .: "val" >>= (.: "kind")
+    let slotP = v .: "val" >>= (.: "slot")
+    case kind' of
+      "TraceNodeIsLeader" -> TraceNode True kind' <$> slotP
+      "TraceNodeNotLeader" -> TraceNode False kind' <$> slotP
+      _ -> fail $ "Expected kind was TraceNodeIsLeader, found " <> show kind' <> "instead"
 
 instance FromJSON Kind where
   parseJSON = Aeson.withObject "Kind" $ \v ->
     Kind <$> v .: "kind"
 
-getRelevantLeaderSlots :: FilePath -> Int -> H.PropertyT (ReaderT IntegrationState (ResourceT IO)) [Int]
-getRelevantLeaderSlots poolNodeStdoutFile slotLowerBound = do
+getRelevantSlots :: FilePath -> Int -> H.PropertyT (ReaderT IntegrationState (ResourceT IO)) ([Int], [Int])
```

Comment:
Returning something like `data LeadershipSlots =  LeadershipSlots { leaderSlots :: [Int], notLeaderSlots :: [Int] }` would make this more readable.